diff --git a/lib/sources/HTS_reference.rb b/lib/sources/HTS_reference.rb
index cce2e2e..0519052 100644
--- a/lib/sources/HTS_reference.rb
+++ b/lib/sources/HTS_reference.rb
@@ -140,11 +140,9 @@ module Organism
   Organism.claim Organism["Hsa"].hg38["hg38.fa"], :proc do |target|
     CMD.get_tool(:bgzip)
     FileUtils.mkdir_p File.dirname(target) unless File.exist? File.dirname(target)
-    url = "https://storage.googleapis.com/genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.fasta"
+    url = "https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz"
     target.sub!(/\.gz$/,'')
-    CMD.cmd_log("wget '#{url}' -O - | bgzip > #{target}.gz")
-    url = "https://storage.googleapis.com/genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.fasta.64.alt"
-    CMD.cmd_log("wget '#{url}' -O #{target}.alt")
+    CMD.cmd_log("wget '#{url}' -O - | gunzip -c | bgzip > #{target}.gz")
     nil
   end
 
diff --git a/workflow.rb b/workflow.rb
index b1b18ea..f2d675d 100644
--- a/workflow.rb
+++ b/workflow.rb
@@ -46,7 +46,10 @@ module HTS
 
     ref = case reference.sub('_noalt','')
           when 'hg19', 'hg38', 'b37', 'hs37d5'
-            Organism["Hsa"][reference][reference + ".fa.gz"]
+            organism_ref = Organism["Hsa"][reference]
+            fasta = organism_ref[reference + ".fa"]
+            fasta_gz = organism_ref[reference + ".fa.gz"]
+            fasta.exists? ? fasta : fasta_gz
           when 'mm10', 'GRCm38'
             Organism["Mmu"].GRCm38["GRCm38.fa.gz"]
           when 'rn6', 'Rno_6.0'
